### PR TITLE
DTSPB-5562: Fix IHT400 journey dead end and allow answer amendment

### PR DESCRIPTION
**Pull request by sadaque-khan-hmcts**
### JIRA link (if applicable)
See [DTSPB-5562](https://tools.hmcts.net/jira/browse/DTSPB-5562)

---

### Change description
Enforce correct IHT400 and HMRC journey state handling to prevent users becoming stuck and ensure valid completion and recovery behaviour.

This PR:
- Enforces strict dependency between IHT400 submission and HMRC probate code steps so completion only occurs for valid end states
- Routes users returning via the task list back to the IHT400 decision point when HMRC conditions are incomplete, enabling answer correction
- Clears dependent HMRC context fields (e.g. `hmrcLetterId`, `uniqueProbateCodeId`) when upstream answers change to prevent stale or inconsistent data
- Refines completion logic to require either no IHT400 requirement or confirmed HMRC letter receipt before marking the journey as complete
- Adds and enhances unit tests to cover IHT400/HMRC state combinations, completion criteria, context reset behaviour, and edge cases, with improved test clarity

This change is **functional**, with supporting refactor and comprehensive test updates.

---

### Testing done
- Manual testing (AAT) across full IHT400 → HMRC journey including complete and incomplete state scenarios
- Verified task list return correctly resolves to the IHT400 decision point when HMRC state is unresolved
- Confirmed upstream answer changes reset dependent HMRC state and prevent dead-end progression
- Unit testing covering all IHT400 and HMRC state combinations (not required, required with letter received, required without letter, unanswered)
- Verified completion only occurs for valid terminal states and remains incomplete otherwise
- All tests pass locally

---

### Impact
Low risk change scoped to IHT journey logic; improves journey integrity, prevents invalid completion states, and guarantees a recoverable path for users to correct incorrect branching decisions.

---

### Security Vulnerability Assessment

**CVE Suppression:** Are there any CVEs present in the codebase (either newly introduced or pre-existing) that are being intentionally suppressed or ignored by this commit?
- [ ] Yes
- [x] No

---

### Pre‑PR checks (must be satisfied)
- [x] Commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation have been updated or added where needed
- [x] Tests have been updated or new tests added where appropriate
- [x] CSS version has been updated to force cache refresh if styling changes were introduced

---

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x] No

---

### Suggested labels
documentation, security, non-functional

### CODE REVIEW

File: app/steps/ui/iht/newsubmittedtohmrc/index.js [hunk 1]
```diff
@@ -30,12 +30,16 @@ class NewSubmittedToHmrc extends ValidationStep {
...
             delete ctx.grossValueField;
             delete ctx.netValueField;
+            delete ctx.hmrcLetterId;
+            delete ctx.uniqueProbateCodeId;
         }
         return super.handlePost(ctx, errors);
     }
     isComplete(ctx) {
+        const noIht400Required = ctx.estateValueCompleted === 'optionNo';
+        const iht400RequiredAndCodeReceived = ctx.estateValueCompleted === 'optionYes' && ctx.hmrcLetterId === 'optionYes';
```
Comment (tom-saunders-cts) [inline app/steps/ui/iht/newsubmittedtohmrc/index.js:40]: i'm not sure whether we can rely on the value in the ctx here for the next question - it may only be set once that page's getContext (or wtte?) method is called and populated, and this may not be the case if we're loading this fresh

File: app/steps/ui/iht/hmrcletter/index.js [hunk 2]
```diff
@@ -15,6 +15,13 @@ class HmrcLetter extends ValidationStep {
         };
     }
 
+    handlePost(ctx, errors) {
+        if (ctx.hmrcLetterId === 'optionNo') {
+            delete ctx.uniqueProbateCodeId;
```
Comment (tom-saunders-cts) [inline app/steps/ui/iht/hmrcletter/index.js:20]: is this needed? where's the value come from?

Comment (sadaque-khan-hmcts) [reply app/steps/ui/iht/hmrcletter/index.js:20]: uniqueProbateCodeId comes from the /unique-probate-code page. The delete is defensive cleanup. If the user changes HMRC letter to No, any previously entered probate code no longer applies, so it should be removed in the IHT sessio ? It also follows the existing IHT pattern were changing the upstream answer clear stale downstream values.

File: app/steps/ui/iht/newsubmittedtohmrc/index.js (hunk 1 above)
Comment (sadaque-khan-hmcts) [reply app/steps/ui/iht/newsubmittedtohmrc/index.js:40]: I believe we can rely on it as task list flow calls getContextData() befofre isComplete, and getContextData builds the ctx from session.form.iht  - check the while loop in step-util js and step js getContextData method. So if the user has already answered /hmrc-letter, hmrcLetterId will be available. If it is missing that means HMRC letter question is unresolved, so /new-submitted-tohmrc should remain incomplete.

File: app/steps/ui/iht/estateform/index.js [hunk 3]
```diff
@@ -15,6 +16,18 @@ class IhtEstateForm extends ValidationStep {
             ]
         };
     }
+
+    isComplete(ctx, formdata) {
+        const dod = formdata && formdata.deceased && formdata.deceased['dod-date'];
```
Comment (tom-saunders-cts) [inline app/steps/ui/iht/estateform/index.js]: this might be clearer as `formdata?.deceased?['dod-date'];` ?

File: app/steps/ui/iht/estateform/index.js [hunk 4]
```diff
@@ -15,6 +16,18 @@ class IhtEstateForm extends ValidationStep {
...
+        const isDodBeforeThreshold = dod ? ExceptedEstateDod.beforeEeDodThreshold(dod) : false;
+        const selectedIht400WithoutHmrcYes = ctx.ihtFormId === 'optionIHT400' && ctx.hmrcLetterId !== 'optionYes';
+
+        if (isDodBeforeThreshold && selectedIht400WithoutHmrcYes) {
```
Comment (tom-saunders-cts) [inline app/steps/ui/iht/estateform/index.js]: this is probably an obvious question but if the dod is after the ee threshold then do we actually end up on this page?

Comment (jamiealbertelli) [reply app/steps/ui/iht/estateform/index.js]: We do not, dates before ee end up
/certificate-interim -> /estate-form 

and dates after ee end up 
/certificate-interim  -> /calc-check -> /new-submitted-to-hmrc

**Review by tom-saunders-cts — APPROVED** [review]
